library/src/main/java/io/github/boguszpawlowski/composecalendar/month/MonthContent.kt

```diff
@@ -112,3 +118,31 @@ internal fun <T : SelectionState> MonthContent(
     }
   }
 }
+
+/**
+ *
+ * */
+@OptIn(ExperimentalPagerApi::class)
+@Composable
+internal fun <T : SelectionState> WeekModeContent(
+  modifier: Modifier = Modifier,
+  selectionState: T,
+  firstDayOfWeek: DayOfWeek = WeekFields.of(Locale.getDefault()).firstDayOfWeek,
+  currentMonth: YearMonth,
+  today: LocalDate = LocalDate.now(),
+  dayContent: @Composable BoxScope.(DayState<T>) -> Unit = { DefaultDay(it) },
+  monthContainer: @Composable (content: @Composable (PaddingValues) -> Unit) -> Unit,
+) {
+  val weeks = currentMonth.getDays(firstDayOfTheWeek = firstDayOfWeek, today = today)
+  Column(
+    modifier = modifier
+  ) {
+    monthContainer {
+      LazyRow(content = {
+        items(weeks) { week ->
+          WeekContent(week = week, selectionState = selectionState, dayContent = dayContent)
+        }
+      })
+    }
+  }
+}
```

Comment:
This is not how this feature should work. Right now, it's a list of all days in a current month. There is no possibility to scroll to next month - the events from scrolling should be passed back to the state of the calendar to force month recalculation. Additionally, scrolling on the weeks itself should work similarly to how the month scrolling works - with a flinging behavior that would stick to the next week. Right now, it's more of a `LinearMode` than a `WeekMode`.